semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/SemanticdbVisitor.java

```diff
@@ -103,8 +106,11 @@ public Void visitClass(ClassTree node, Void unused) {
   public Void visitMethod(MethodTree node, Void unused) {
     if (node instanceof JCTree.JCMethodDecl) {
       JCTree.JCMethodDecl meth = (JCTree.JCMethodDecl) node;
-      emitSymbolOccurrence(
-          meth.sym, meth, Role.DEFINITION, CompilerRange.FROM_POINT_TO_SYMBOL_NAME);
+      CompilerRange range = CompilerRange.FROM_POINT_TO_SYMBOL_NAME;
+      if (meth.params.isEmpty() && (meth.sym.flags() & Flags.GENERATEDCONSTR) != 0L) {
```

Comment:
Minor: might be worth commenting why this is needed here. It looks like we're specifically targeting only constructors? Is there a difference between this and `name.contentMatches("<init>")`? Is it possible to reuse the logic between this line and the one in `RangeFinder`?

Reply:
The difference being, my thinking at the time was that only the no-param constructor would ever be generated, but I realise now this doesn't hold true for the new `record` class types, so I'll revisit this line.

Reply:
We can BTW add Java 15 code to the test suite. Alternatively, we can change the default minimized project to Java 15

Reply:
I wouldn't worry too much about Java 15 however, we can deal with it later. I suspect very few users are on Java 15 right now.